micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java

```diff
@@ -674,6 +654,32 @@ public More more() {
         return (M) m;
     }
 
+    private Meter getOrCreateMeter(HistogramConfig config, BiFunction<Id, HistogramConfig, Meter> builder, Id mappedId) {
+        Meter m = meterMap.get(mappedId);
+
+        if (m == null) {
+            synchronized (this) {
+                m = meterMap.get(mappedId);
```

Comment:
`meterMap` needs to be `volatile`, at least.

Reply:
Sure

Reply:
@padilo please make one of:
- Add to javadocs that MeterRegistry does synchronization by itself instead of private non-shared lock, it is good practice to always document that synchronization performed on "this" because users will be better informed about potential deadlocks.
- Or just use private non-shared lock.

Reply:
+1 for private non-shared lock